### ISSUE TITLE
[Bug] [move-compiler-v2] aptos-move compile --compiler-version=v2 doesn't have correct attributes

### ISSUE THREAD

**Issue by brmataptos** [aptos-labs/aptos-core#10896]
# 🐛 Bug

We should not be seeing warnings for attributes like `#[view]` on aptos code:

```
warning: unknown attribute
     ┌─ /Users/tengzhang/aptos-core/aptos-move/framework/aptos-token/sources/token.move:1521:7
     │
1521 │     #[view]
     │       ^^^^ Attribute name 'view' is unknown (use --skip-attribute-checks CLI option to ignore); known attributes are '{"bytecode_instruction", "deprecated", "expected_failure", "native_interface", "test", "test_only", "verify_only"}'.
```

**Comment by brmataptos** [comment]
Note that the attributes are lost (along with other flags) in third_party/move/tools/move-package/src/compilation.rs, in `CompiledPackage::build_all` when `flags` are ignored in the `CompilerVersion::V2` clause around line 635.

**Comment by brmataptos** [comment]
Here's a backtrace, from running `aptos move compile --compiler-version=v2` in aptos-move/framework/aptos-framework with some debugging enabled:

```
Backtrace: Backtrace [
    { fn: "std::backtrace_rs::backtrace::libunwind::trace", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/../../backtrace/src/backtrace/libunwind.rs", line: 93 },
    { fn: "std::backtrace_rs::backtrace::trace_unsynchronized", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/../../backtrace/src/backtrace/mod.rs", line: 66 },
    { fn: "std::backtrace::Backtrace::create", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/backtrace.rs", line: 332 },
    { fn: "move_compiler::expansion::translate::unique_attributes", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler/src/expansion/translate.rs", line: 743 },
    { fn: "move_compiler::expansion::translate::flatten_attributes", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler/src/expansion/translate.rs", line: 721 },
    { fn: "move_compiler::expansion::translate::function_", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler/src/expansion/translate.rs", line: 1585 },
    { fn: "move_compiler::expansion::translate::function", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler/src/expansion/translate.rs", line: 1565 },
    { fn: "move_compiler::expansion::translate::module_", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler/src/expansion/translate.rs", line: 582 },
    { fn: "move_compiler::expansion::translate::module", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler/src/expansion/translate.rs", line: 421 },
    { fn: "move_compiler::expansion::translate::definition", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler/src/expansion/translate.rs", line: 296 },
    { fn: "move_compiler::expansion::translate::program", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler/src/expansion/translate.rs", line: 212 },
    { fn: "move_compiler::command_line::compiler::run", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler/src/command_line/compiler.rs", line: 789 },
    { fn: "move_compiler::command_line::compiler::SteppedCompiler<_>::run_impl", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler/src/command_line/compiler.rs", line: 283 },
    { fn: "move_compiler::command_line::compiler::SteppedCompiler<_>::run", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler/src/command_line/compiler.rs", line: 392 },
    { fn: "move_model::run_model_builder_with_options_and_compilation_flags", file: "/Users/brm/code/aptos-core/third_party/move/move-model/src/lib.rs", line: 233 },
    { fn: "move_model::run_model_builder_in_compiler_mode", file: "/Users/brm/code/aptos-core/third_party/move/move-model/src/lib.rs", line: 92 },
    { fn: "move_compiler_v2::run_checker", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler-v2/src/lib.rs", line: 96 },
    { fn: "move_compiler_v2::run_move_compiler", file: "/Users/brm/code/aptos-core/third_party/move/move-compiler-v2/src/lib.rs", line: 48 },
    { fn: "move_package::compilation::compiled_package::build_and_report_no_exit_v2_driver", file: "/Users/brm/code/aptos-core/third_party/move/tools/move-package/src/compilation/compiled_package.rs", line: 1077 },
    { fn: "core::ops::function::FnMut::call_mut", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/ops/function.rs", line: 166 },
    { fn: "move_package::compilation::compiled_package::CompiledPackage::build_all", file: "/Users/brm/code/aptos-core/third_party/move/tools/move-package/src/compilation/compiled_package.rs", line: 673 },
    { fn: "move_package::compilation::build_plan::BuildPlan::compile_with_driver", file: "/Users/brm/code/aptos-core/third_party/move/tools/move-package/src/compilation/build_plan.rs", line: 197 },
    { fn: "move_package::compilation::build_plan::BuildPlan::compile_no_exit", file: "/Users/brm/code/aptos-core/third_party/move/tools/move-package/src/compilation/build_plan.rs", line: 131 },
    { fn: "move_package::BuildConfig::compile_package_no_exit", file: "/Users/brm/code/aptos-core/third_party/move/tools/move-package/src/lib.rs", line: 214 },
    { fn: "aptos_framework::built_package::BuiltPackage::build", file: "/Users/brm/code/aptos-core/aptos-move/framework/src/built_package.rs", line: 183 },
    { fn: "<aptos::move_tool::CompilePackage as aptos::common::types::CliCommand<alloc::vec::Vec<alloc::string::String>>>::execute::{{closure}}", file: "/Users/brm/code/aptos-core/crates/aptos/src/move_tool/mod.rs", line: 319 },
    { fn: "<core::pin::Pin<P> as core::future::future::Future>::poll", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/future/future.rs", line: 125 },
    { fn: "aptos::common::types::CliCommand::execute_serialized_with_logging_level::{{closure}}", file: "/Users/brm/code/aptos-core/crates/aptos/src/common/types.rs", line: 1210 },
    { fn: "<core::pin::Pin<P> as core::future::future::Future>::poll", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/future/future.rs", line: 125 },
    { fn: "aptos::common::types::CliCommand::execute_serialized::{{closure}}", file: "/Users/brm/code/aptos-core/crates/aptos/src/common/types.rs", line: 1198 },
    { fn: "<core::pin::Pin<P> as core::future::future::Future>::poll", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/future/future.rs", line: 125 },
    { fn: "aptos::move_tool::MoveTool::execute::{{closure}}", file: "/Users/brm/code/aptos-core/crates/aptos/src/move_tool/mod.rs", line: 104 },
    { fn: "aptos::Tool::execute::{{closure}}", file: "/Users/brm/code/aptos-core/crates/aptos/src/lib.rs", line: 67 },
    { fn: "tokio::runtime::park::CachedParkThread::block_on::{{closure}}", file: "/Users/brm/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/park.rs", line: 283 },
    { fn: "tokio::runtime::coop::with_budget", file: "/Users/brm/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/coop.rs", line: 107 },
    { fn: "tokio::runtime::coop::budget", file: "/Users/brm/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/coop.rs", line: 73 },
    { fn: "tokio::runtime::park::CachedParkThread::block_on", file: "/Users/brm/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/park.rs", line: 283 },
    { fn: "tokio::runtime::context::blocking::BlockingRegionGuard::block_on", file: "/Users/brm/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/context/blocking.rs", line: 66 },
    { fn: "tokio::runtime::scheduler::multi_thread::MultiThread::block_on::{{closure}}", file: "/Users/brm/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/scheduler/multi_thread/mod.rs", line: 87 },
    { fn: "tokio::runtime::context::runtime::enter_runtime", file: "/Users/brm/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/context/runtime.rs", line: 65 },
    { fn: "tokio::runtime::scheduler::multi_thread::MultiThread::block_on", file: "/Users/brm/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/scheduler/multi_thread/mod.rs", line: 86 },
    { fn: "tokio::runtime::runtime::Runtime::block_on", file: "/Users/brm/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/runtime.rs", line: 313 },
    { fn: "aptos::main", file: "/Users/brm/code/aptos-core/crates/aptos/src/main.rs", line: 27 },
    { fn: "core::ops::function::FnOnce::call_once", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/ops/function.rs", line: 250 },
    { fn: "std::sys_common::backtrace::__rust_begin_short_backtrace", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/sys_common/backtrace.rs", line: 135 },
    { fn: "std::rt::lang_start::{{closure}}", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/rt.rs", line: 166 },
    { fn: "core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/ops/function.rs", line: 284 },
    { fn: "std::panicking::try::do_call", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/panicking.rs", line: 500 },
    { fn: "std::panicking::try", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/panicking.rs", line: 464 },
    { fn: "std::panic::catch_unwind", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/panic.rs", line: 142 },
    { fn: "std::rt::lang_start_internal::{{closure}}", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/rt.rs", line: 148 },
    { fn: "std::panicking::try::do_call", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/panicking.rs", line: 500 },
    { fn: "std::panicking::try", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/panicking.rs", line: 464 },
    { fn: "std::panic::catch_unwind", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/panic.rs", line: 142 },
    { fn: "std::rt::lang_start_internal", file: "/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/rt.rs", line: 148 },
    { fn: "_main" },
]
```

**Comment by rahxephon89** [comment]
Another interesting observation from by side is that when using `aptos move compile --compiler-version=v2  --skip-attribute-checks`,  the warning messages still show up. Wondering whether it is expected. cc @brmataptos

**Comment by brmataptos** [comment]
Seems to be fixed by #11610 .
